pkg/distributor/distributor.go

```diff
@@ -300,6 +305,25 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	}
 }
 
+func (d *Distributor) truncateLines(vContext validationContext, stream logproto.Stream) []logproto.Entry {
+	if !vContext.maxLineSizeShouldTruncate {
+		return stream.Entries
+	}
+
+	n := 0
+	for _, e := range stream.Entries {
+		if maxSize := vContext.maxLineSize; maxSize != 0 && len(e.Line) > maxSize {
+			truncated := float64(len(e.Line) - maxSize)
+			stream.Entries[n].Line = e.Line[:maxSize]
+			n++
```

Comment:
It looks you're just using n to track which entry in `stream.Entries` to truncate and replace? In that case you can just do `for i, e := range stream.Entries` and where you have `stream.Entries[n].Line`.

Reply:
It looks like n is being used to increment the number of truncated lines, which is then added to a metric.

Reply:
This wasn't being used in a metric. I shamelessly copied this pattern without thinking from  the `Push` method to avoid a new slice but didn't catch why it was used there.